### ISSUE TITLE
Running a workflow and only first steps are executed

### ISSUE THREAD

**Issue by yvanlebras** [galaxyproject/galaxy#4127]
I have encountered some issues on a Galaxy 17.05 test server and have seen the same behaviour on usegalaxy.org. Some workflows are not executed completely e.g. only first 4 steps for example are executed and nothing (apparently no queued jobs so not the same issue than  #3816) for the last X steps.... 

Has anyone seen this as well?

**Comment by mvdbeek** [comment]
It would be very helpful if you could share a workflow demonstrating this problem.

**Comment by yvanlebras** [comment]
After making some tests, I've found the problem! This is related to my other issue #4128. As I was thinking that tagging a dataset through a workflow step "configure output" option was not working, I have made tests and one was to tag with a `#tag,` term instead of the classical `#tag` syntax.... So it's due to an improper use of tag expression!


![image](https://cloud.githubusercontent.com/assets/7910679/26627135/b2bf6a46-45f9-11e7-8cec-0493bde08cf9.png)

**Comment by mvdbeek** [comment]
That's not exactly the same issue though, here the workflow fails to schedule, while the other issue is that you don't get these colorful tags.

FWIW, I can reproduce this (with a trailing `,` in the hashtag), this is the traceback:
```
galaxy.workflow.run ERROR 2017-05-31 12:17:32,494 Failed to execute scheduled workflow.
Traceback (most recent call last):
  File "/Users/mvandenb/src/galaxy/lib/galaxy/workflow/run.py", line 80, in __invoke
    outputs = invoker.invoke()
  File "/Users/mvandenb/src/galaxy/lib/galaxy/workflow/run.py", line 169, in invoke
    jobs = self._invoke_step( step )
  File "/Users/mvandenb/src/galaxy/lib/galaxy/workflow/run.py", line 241, in _invoke_step
    jobs = step.module.execute( self.trans, self.progress, self.workflow_invocation, step )
  File "/Users/mvandenb/src/galaxy/lib/galaxy/workflow/modules.py", line 849, in execute
    self._handle_post_job_actions( step, job, invocation.replacement_dict )
  File "/Users/mvandenb/src/galaxy/lib/galaxy/workflow/modules.py", line 914, in _handle_post_job_actions
    ActionBox.execute( self.trans.app, self.trans.sa_session, pja, job, replacement_dict )
  File "/Users/mvandenb/src/galaxy/lib/galaxy/jobs/actions/post.py", line 397, in execute
    ActionBox.actions[pja.action_type].execute(app, sa_session, pja, job, replacement_dict)
  File "/Users/mvandenb/src/galaxy/lib/galaxy/jobs/actions/post.py", line 337, in execute
    app.tag_handler.set_tags_from_list( job.user, dataset_assoc.dataset, tags)
  File "/Users/mvandenb/src/galaxy/lib/galaxy/managers/tags.py", line 46, in set_tags_from_list
    self.apply_item_tags( user, item, unicodify( new_tags_str, 'utf-8' ) )
  File "/Users/mvandenb/src/galaxy/lib/galaxy/managers/tags.py", line 165, in apply_item_tags
    self.apply_item_tag( user, item, name, value )
  File "/Users/mvandenb/src/galaxy/lib/galaxy/managers/tags.py", line 131, in apply_item_tag
    lc_name = name.lower()
AttributeError: 'NoneType' object has no attribute 'lower'
```

So we should do some additional validation for the correct syntax of tags.

**Comment by dannon** [comment]
(replied in #4128, I'm taking a look at this workflow tag editor this morning)

**Comment by mvdbeek** [comment]
Closing this as it has been fixed in #4139. Feel free to re-open if the problem pops up again.
